packages/bvaughn-architecture-demo/components/console/hooks/useConsoleSearchDOM.ts

```diff
@@ -0,0 +1,82 @@
+import useSearchDOM from "@bvaughn/src/hooks/useSearchDOM";
+import type {
+  Actions as SearchActions,
+  State as SearchState,
+} from "@bvaughn/src/hooks/useSearchDOM";
+import { MutableRefObject, useMemo, useState } from "react";
+
+import useFilteredMessagesDOM, { Loggable } from "./useFilteredMessagesDOM";
+
+const EMPTY_ARRAY: any[] = [];
+
+function search(
+  query: string,
+  loggables: Loggable[],
+  listRef: MutableRefObject<HTMLElement | null>
+): Loggable[] {
+  const results: Loggable[] = [];
+  const needle = query.toLocaleLowerCase();
+
+  const list = listRef.current!;
+  list.childNodes.forEach((node: ChildNode, index: number) => {
+    const element = node as HTMLElement;
+
+    // HACK Must be compatible with the style used by useFilteredMessagesDOM()
+    if (element.style.display != "none") {
```

Comment:
This coupling feels a little yuck and I'm open for suggestions if anyone has them.